packages/web3torrent/src/pages/file/File.tsx

```diff
@@ -53,10 +55,28 @@ const File: React.FC<RouteComponentProps & Props> = props => {
       </div>
       <WebTorrentContext.Consumer>
         {web3Torrent => {
-          const me = web3Torrent.paymentChannelClient.mySigningAddress;
-          const channelCache = web3Torrent.paymentChannelClient.channelCache;
+          const paymentChannelClient = web3Torrent.paymentChannelClient;
+          const me = paymentChannelClient.mySigningAddress;
+          const channelCache = paymentChannelClient.channelCache;
+          const budgetCache = paymentChannelClient.budgetCache;
+
+          // Only show budget when any channel exists.
+          const showBudget =
+            Object.keys(budgetCache).length > 0 && Object.keys(channelCache).length > 0;
           return (
-            <TorrentInfo torrent={torrent} channelCache={channelCache} mySigningAddress={me} />
+            <>
+              <TorrentInfo torrent={torrent} channelCache={channelCache} mySigningAddress={me} />
+              <br />
+              {showBudget ? (
+                <SiteBudgetTable
+                  budgetCache={budgetCache as SiteBudget}
```

Comment:
Is this type assertion necessary?

Reply:
Yeah. We have in the most places defined `SiteBudget | {}` or `SiteBudget | undefined` as the return type.